core/templates/dev/head/domain/exploration/ReadOnlyExplorationBackendApiService.js

```diff
@@ -0,0 +1,163 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to retrieve read only information
+ * about explorations from the backend.
+ */
+
+oppia.factory('ReadOnlyExplorationBackendApiService', [
+  '$http', '$q', 'EXPLORATION_DATA_URL_TEMPLATE',
+  'EXPLORATION_VERSION_DATA_URL_TEMPLATE', 'UrlInterpolationService',
+  function($http, $q, EXPLORATION_DATA_URL_TEMPLATE,
+    EXPLORATION_VERSION_DATA_URL_TEMPLATE, UrlInterpolationService) {
+    // Maps previously loaded explorations to their IDs.
+    var _explorationCache = [];
+
+    var _fetchExploration = function(
+        explorationId, version, successCallback, errorCallback) {
+      var explorationDataUrl = _getExplorationUrl(explorationId, version);
+
+      $http.get(explorationDataUrl).then(function(response) {
+        var exploration = angular.copy(response.data);
+        if (successCallback) {
+          successCallback(exploration);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _isCached = function(explorationId) {
+      return _explorationCache.hasOwnProperty(explorationId);
+    };
+
+    var _getExplorationUrl = function(explorationId, version) {
+      if (version) {
+        return UrlInterpolationService.interpolateUrl(
+        EXPLORATION_VERSION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId,
+          v: String(version)
+        });
+      }
+      return UrlInterpolationService.interpolateUrl(
+        EXPLORATION_DATA_URL_TEMPLATE, {
+          exploration_id: explorationId
+        }
+      );
+    };
+
+    return {
+      /**
+       * Retrieves an exploration from the backend given an exploration ID
+       * and version number (or none). This returns a promise object that
+       * allows success and rejection callbacks to be registered. If the
+       * exploration is successfully loaded and a success callback function
+       * is provided to the promise object, the success callback is called
+       * with the exploration passed in as a parameter. If something goes
+       * wrong while trying to fetch the exploration, the rejection callback
+       * is called instead, if present. The rejection callback function is
+       * passed an error message using this format
```

Comment:
How do we guarantee this format? From the earlier code, it seems we just pass whatever the backend tells us is the error. Is that not true?